posthog/models/project.py

```diff
@@ -0,0 +1,38 @@
+from typing import TYPE_CHECKING, Optional, Tuple
+from django.db import models
+from django.db import transaction
+from django.core.validators import MinLengthValidator
+
+if TYPE_CHECKING:
+    from .team import Team
+
+
+class ProjectManager(models.Manager):
+    def create_with_team(self, team_fields: Optional[dict] = None, **kwargs) -> Tuple["Project", "Team"]:
+        from .team import Team
+
+        with transaction.atomic():
+            common_id = Team.objects.increment_id_sequence()
+            project = self.create(id=common_id, **kwargs)
+            team = Team.objects.create(
+                id=common_id, organization=project.organization, project=project, **(team_fields or {})
+            )
+            return project, team
+
+
+class Project(models.Model):
```

Comment:
Nit: Maybe we should add a comment here warning and explaining that this is part of a longer term plan? Thinking for newer people / people not super clued up here that they could accidentally start using it.

Reply:
I do always appreciate a docstring. Added